x-pack/elastic-agent/pkg/composable/providers/kubernetes_secrets/kubernetes_secrets.go

```diff
@@ -0,0 +1,88 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package kubernetes_secrets
+
+import (
+	"context"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "k8s.io/client-go/kubernetes"
+
+	"github.com/elastic/beats/v7/libbeat/common/kubernetes"
+	"github.com/elastic/beats/v7/libbeat/keystore"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/agent/errors"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/composable"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/config"
+	corecomp "github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/core/composable"
+	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/core/logger"
+)
+
+func init() {
+	composable.Providers.AddContextProvider("kubernetes_secrets", ContextProviderBuilder)
+}
+
+type contextProviderK8sSecrets struct {
+	logger *logger.Logger
+	config *Config
+	client k8sclient.Interface
+}
+
+// DynamicProviderBuilder builds the dynamic provider.
+func ContextProviderBuilder(logger *logger.Logger, c *config.Config) (corecomp.ContextProvider, error) {
+	var cfg Config
+	if c == nil {
+		c = config.New()
+	}
+	err := c.Unpack(&cfg)
+	if err != nil {
+		return nil, errors.New(err, "failed to unpack configuration")
+	}
+	return &contextProviderK8sSecrets{logger, &cfg, nil}, nil
+}
+
+func (p *contextProviderK8sSecrets) Fetch(key string) (string, error) {
```

Comment:
This always seems to be the same error, could this just be changed to `string, bool` in the return? That way the code above can just check for `bool` and return the `ErrNoMatch`.